### ISSUE TITLE
Granting access to a user should not first revoke access

### ISSUE THREAD

**Issue by sebastienpattyn93** [juju/python-libjuju#173]
I noticed that granting model access to user is first calling the revoke method.
```
    async def grant(self, username, acl='read'):
        """Grant a user access to this model.
        :param str username: Username
        :param str acl: Access control ('read' or 'write')
        """
        controller_conn = await self.connection.controller()
        model_facade = client.ModelManagerFacade.from_connection(
            controller_conn)
        user = tag.user(username)
        model = tag.model(self.info.uuid)
        changes = client.ModifyModelAccess(acl, 'grant', model, user)
        await self.revoke(username)
        return await model_facade.ModifyModelAccess([changes])
```
The problem is when I created a user who has no modelaccess, this code fails because there is nothing to revoke.

```
2017-10-12 10:36:19,511 ERROR   File "/opt/sojobo_api/scripts/set_model_access.py", line 47, in set_model_acc
    await model.grant(user, acl='write')

2017-10-12 10:36:19,511 ERROR   File "/usr/lib/python3.6/asyncio/coroutines.py", line 109, in __next__
    return self.gen.send(None)

2017-10-12 10:36:19,511 ERROR   File "/usr/local/lib/python3.6/dist-packages/juju/model.py", line 1293, in grant
    await self.revoke(username)

2017-10-12 10:36:19,511 ERROR   File "/usr/lib/python3.6/asyncio/coroutines.py", line 109, in __next__
    return self.gen.send(None)

2017-10-12 10:36:19,511 ERROR   File "/usr/local/lib/python3.6/dist-packages/juju/model.py", line 1442, in revoke
    return await model_facade.ModifyModelAccess([changes])

2017-10-12 10:36:19,511 ERROR   File "/usr/lib/python3.6/asyncio/coroutines.py", line 109, in __next__
    return self.gen.send(None)

2017-10-12 10:36:19,511 ERROR   File "/usr/local/lib/python3.6/dist-packages/juju/client/facade.py", line 392, in wrapper
    reply = await f(*args, **kwargs)

2017-10-12 10:36:19,511 ERROR   File "/usr/lib/python3.6/asyncio/coroutines.py", line 109, in __next__
    return self.gen.send(None)

2017-10-12 10:36:19,511 ERROR   File "/usr/local/lib/python3.6/dist-packages/juju/client/_client3.py", line 2518, in ModifyModelAccess
    reply = await self.rpc(msg)

2017-10-12 10:36:19,511 ERROR   File "/usr/lib/python3.6/asyncio/coroutines.py", line 109, in __next__
    return self.gen.send(None)

2017-10-12 10:36:19,512 ERROR   File "/usr/local/lib/python3.6/dist-packages/juju/client/facade.py", line 514, in rpc
    result = await self.connection.rpc(msg, encoder=TypeEncoder)

2017-10-12 10:36:19,512 ERROR   File "/usr/lib/python3.6/asyncio/coroutines.py", line 109, in __next__
    return self.gen.send(None)

2017-10-12 10:36:19,512 ERROR   File "/usr/local/lib/python3.6/dist-packages/juju/client/connection.py", line 280, in rpc
    raise JujuError(errors)

2017-10-12 10:36:19,512 ERROR juju.errors.JujuError: ['could not revoke model access: model user "trialaccount" does not exist']
```

There should be a check if the user already has a certain access level or not. Or it should be removed and we would have to do the check if a revoke is needed.

**Comment by sebastienpattyn93** [comment]
I was looking for a possible solution to fix this but I noticed that the ModelManagerFacede has the property `ModifyModelAccess` but not the possibility to retrieve ModelAccess while in the `ControllerFacade` I noticed that `ModifyControllerAccess` and `GetControllerAccess` both exists. Is that normal because now there is no direct way to get the model access to check if the user has access to a model if not. This way it is hard to check if access will have to be revoked or not.
